source/extensions/priv/server/ntds_decrypt.c

```diff
@@ -0,0 +1,129 @@
+#include "precomp.h"
+
+void bytes_to_string(LPBYTE data, int length, LPSTR output){
+	for (int i = 0; i < length; i++){
+		sprintf(output + (i << 1), "%02X", ((LPBYTE)data)[i]);

```

Comment:
I'd prefer to see some kind of bounds checking on `output` so that we don't make it possible for an overflow. Use of `snprintf_s` would be good.